x/authz/module/abci.go

```diff
@@ -0,0 +1,20 @@
+package authz
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz/keeper"
+)
+
+// EndBlocker is called at the end of every block
+func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
```

Comment:
I think this should be done in `BeginBlocker`

Reply:
I was following `gov` proposal implementation. We are deleting them in the [EndBlocker](https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/abci.go#L15).

Reply:
I  think we should prune at the beginning - there is no point to potentially keep them during the tx execution. The following can be true:
* in block `B_n`, a grant can have `g.expire > B_n.BlockTime()`
* but in the next block `B_n+1` it could be: `g.expire < B_{n+1}.BlockTime()`.